pyproject.toml

```diff
@@ -42,6 +42,8 @@ dependencies = [
     'pooch>=1.7.0',
     'pycairo>=1.20; sys_platform == "win32"',
     'joblib>=1.3.1',
+    'palmotif',
+    'IPython',
```

Comment:
Why is the IPython dependency necessary?

Reply:
Palmotif returns a SVG object, which is by default saved as such accordingly. What I did here was that I used the IPython.display's function SVG to directly display the SVG in my notebook and not always have to save it as a file. This was a workaround that I have almost forgotten, and I am not sure that this is even necessary...although I like to be able to directly investigate plots inside the notebook. However, I noticed that this is probably not the smartest way to do as I do not offer the possibility to save the SVG to a file any more and I might adapt this accordingly. Either way, I am not sure if SVG is a handy format for users to deal with... do you know if there is a possibility to save it as a png file or any other format that might be more convenient to deal with?

Reply:
Will have a look into this, but sounds promising :D

Reply:
I am just having a hard time to manipulate the plot as plt.show() is called inside mpl_plot and it returns three objects, which are somewhat confusing to me...

Do you know if a object such as 
[[(Text(0, 0, 'C'), 178.8135375281251)],
 [(Text(0, 0, 'A'), 178.8135375281251)],
 [(Text(0, 0, 'S'), 178.8135375281251)],
 [(Text(0, 0, 'S'), 178.8135375281251)],
 [(Text(0, 0, 'E'), 11.495650387247952),
  (Text(0, 0, 'H'), 11.495650387247952),
  (Text(0, 0, 'L'), 11.495650387247952),
  (Text(0, 0, 'T'), 11.495650387247952),
  (Text(0, 0, 'Y'), 11.495650387247958),
  (Text(0, 0, 'P'), 22.991300774495915)],
 [(Text(0, 0, 'G'), 13.08748712194183),
  (Text(0, 0, 'S'), 13.08748712194183),
  (Text(0, 0, 'Y'), 13.08748712194183),
  (Text(0, 0, 'P'), 26.17497424388366),
  (Text(0, 0, 'V'), 26.17497424388366)],
 [(Text(0, 0, 'G'), 11.495650387247952),
  (Text(0, 0, 'I'), 11.495650387247952),
  (Text(0, 0, 'L'), 11.495650387247952),
  (Text(0, 0, 'S'), 11.495650387247952),
  (Text(0, 0, 'V'), 11.495650387247958),
  (Text(0, 0, 'W'), 22.991300774495915)],
 [(Text(0, 0, 'F'), 17.862997326023464),
  (Text(0, 0, 'A'), 35.72599465204693),
  (Text(0, 0, 'G'), 71.45198930409386)],
 [(Text(0, 0, 'A'), 13.688315950194387),
  (Text(0, 0, 'E'), 13.688315950194387),
  (Text(0, 0, 'L'), 13.688315950194387),
  (Text(0, 0, 'V'), 13.688315950194387),
  (Text(0, 0, 'G'), 41.06494785058316)],
 [(Text(0, 0, 'A'), 11.49565038724795),
  (Text(0, 0, 'E'), 11.49565038724795),
  (Text(0, 0, 'L'), 11.49565038724795),
  (Text(0, 0, 'P'), 11.49565038724795),
  (Text(0, 0, 'V'), 11.495650387247956),
  (Text(0, 0, 'G'), 22.9913007744959)],
 [(Text(0, 0, 'T'), 14.679323856635706),
  (Text(0, 0, 'G'), 29.358647713271413),
  (Text(0, 0, 'L'), 29.358647713271413),
  (Text(0, 0, 'P'), 29.358647713271402)],
 [(Text(0, 0, 'D'), 13.08748712194183),
  (Text(0, 0, 'I'), 13.08748712194183),
  (Text(0, 0, 'L'), 13.08748712194183),
  (Text(0, 0, 'G'), 26.17497424388366),
  (Text(0, 0, 'S'), 26.17497424388366)],
 [(Text(0, 0, 'N'), 15.280152684888265),
  (Text(0, 0, 'I'), 15.280152684888265),
  (Text(0, 0, 'L'), 30.56030536977653),
  (Text(0, 0, 'S'), 45.840458054664815)],
 [(Text(0, 0, 'L'), 17.472818247834695),
  (Text(0, 0, 'G'), 52.41845474350409),
  (Text(0, 0, 'S'), 52.41845474350409)],
 [(Text(0, 0, 'N'), 13.688315950194387),
  (Text(0, 0, 'Q'), 13.688315950194387),
  (Text(0, 0, 'S'), 13.688315950194387),
  (Text(0, 0, 'T'), 13.688315950194387),
  (Text(0, 0, 'A'), 41.06494785058316)],
 [(Text(0, 0, 'D'), 13.688315950194387),
  (Text(0, 0, 'Q'), 13.688315950194387),
  (Text(0, 0, 'E'), 13.688315950194387),
  (Text(0, 0, 'Y'), 13.688315950194387),
  (Text(0, 0, 'N'), 41.06494785058316)],
 [(Text(0, 0, 'E'), 15.280152684888265),
  (Text(0, 0, 'P'), 15.280152684888265),
  (Text(0, 0, 'T'), 30.56030536977653),
  (Text(0, 0, 'V'), 45.840458054664815)],
 [(Text(0, 0, 'L'), 60.16698866690969), (Text(0, 0, 'Q'), 80.2226515558796)],
 [(Text(0, 0, 'H'), 17.472818247834695),
  (Text(0, 0, 'T'), 52.41845474350409),
  (Text(0, 0, 'Y'), 52.41845474350409)],
 [(Text(0, 0, 'F'), 178.8135375281251)]]
can be easily transformed into a matplotlib figure or if it can be used to customize the returned matplotlib plot?

Reply:
> as plt.show() is called inside mpl_plot 

that's annoying. 

When choosing palmotif, did you also take a look at logomaker?
https://logomaker.readthedocs.io/en/latest/

Neither palmotif nor logomaker seem very actively maintained, but logomaker seems much more popular (according to github stars). From a first glance at the docs, logomaker seems more customizable and I also like that it doesn't have parasail as a hard dependency which I'd like to get rid of soon (see #450). But I'm not sure if it has other limitations.

Reply:
To be honest, I am not sure why I have chosen palmotif...probably because it was used in the single-cell best practice book as well. I will have a look on logomaker, but if it's even preferable to use logomaker over palmotif (to get rid of parasail) I would be happy to adapt my code accordingly unless it is for whatever reason impossible to use logomaker in our case...